### PR TITLE
Enable Safe Browsing endpoint for extension blacklist 

### PR DESCRIPTION
**Pull request by fmarier**
Fixes brave/brave-browser#12297

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [x] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

Here's how I manually tested this PR:

1. Apply this patch:
```
diff --git a/patches/chrome-browser-extensions-blocklist.cc.patch b/patches/chrome-browser-extensions-blocklist.cc.patch
new file mode 100644
index 0000000000..921bba8d24
--- /dev/null
+++ b/patches/chrome-browser-extensions-blocklist.cc.patch
@@ -0,0 +1,43 @@
+diff --git a/chrome/browser/extensions/blocklist.cc b/chrome/browser/extensions/blocklist.cc
+index cc678a66b37040d0c735edd1218baf7b9e435f25..2e81a0922aaefca5a10cf52c185f91cb1f44f3a2 100644
+--- a/chrome/browser/extensions/blocklist.cc
++++ b/chrome/browser/extensions/blocklist.cc
+@@ -133,8 +133,12 @@ class SafeBrowsingClientImpl
+ 
+ void CheckOneExtensionState(Blocklist::IsBlocklistedCallback callback,
+                             const Blocklist::BlocklistStateMap& state_map) {
+-  std::move(callback).Run(state_map.empty() ? NOT_BLOCKLISTED
+-                                            : state_map.begin()->second);
++  BlocklistState state = state_map.empty() ? NOT_BLOCKLISTED : state_map.begin()->second;
++  if (state == BLOCKLISTED_POTENTIALLY_UNWANTED) {
++      state = BLOCKLISTED_MALWARE;
++  }
++  LOG(ERROR) << "state=" << state;
++  std::move(callback).Run(state);
+ }
+ 
+ void GetMalwareFromBlocklistStateMap(
+@@ -219,7 +223,12 @@ void Blocklist::GetMalwareIDs(const std::set<std::string>& ids,
+ void Blocklist::IsBlocklisted(const std::string& extension_id,
+                               IsBlocklistedCallback callback) {
+   std::set<std::string> check;
+-  check.insert(extension_id);
++  if (extension_id == "jknemblkbdhdcpllfgbfekkdciegfboi") {
++    check.insert("didkgikegfpeeegmomledbebilmomnpb");
++    LOG(ERROR) << "REPLACING EXTENSION ID WITH BAD ONE";
++  } else {
++    check.insert(extension_id);
++  }
+   GetBlocklistedIDs(
+       check, base::BindOnce(&CheckOneExtensionState, std::move(callback)));
+ }
+@@ -290,6 +299,9 @@ void Blocklist::RequestExtensionsBlocklistState(
+ void Blocklist::OnBlocklistStateReceived(const std::string& id,
+                                          BlocklistState state) {
+   DCHECK_CURRENTLY_ON(BrowserThread::UI);
++  if (state == BLOCKLISTED_POTENTIALLY_UNWANTED) {
++      state = BLOCKLISTED_MALWARE;
++  }
+   blocklist_state_cache_[id] = state;
+ 
+   // Go through the opened requests and call the callbacks for those requests
```
2. Watch the outgoing network requests using mitmproxy or Charles.
3. Install this extension: https://chrome.google.com/webstore/detail/suspicious-site-reporter/jknemblkbdhdcpllfgbfekkdciegfboi/related
4. You should see a POST (200) request to safebrowsing2.brave.com and this in
the terminal:
```
[3421702:3421702:0205/150902.462491:ERROR:blocklist.cc(228)] REPLACING EXTENSION ID WITH BAD ONE
[3421702:3421702:0205/150902.968802:ERROR:blocklist.cc(140)] state=1
```
5. Open `brave://extensions/` and you should see:
Screenshot from 2021-02-05 15-09-23

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/net/brave_block_safebrowsing_urls.cc [hunk 1]
```diff
@@ -20,6 +20,9 @@ bool IsSafeBrowsingReportingURL(const GURL& gurl) {
       URLPattern(
           URLPattern::SCHEME_HTTPS,
           "https://sb-ssl.google.com/safebrowsing/clientreport/download*"),
+      URLPattern(
+          URLPattern::SCHEME_HTTPS,
+          "https://sb-ssl.google.com/safebrowsing/clientreport/crx-list-info"),
```
Comment (jumde) [inline browser/net/brave_block_safebrowsing_urls.cc:25]: May be add a test in `brave_static_redirect_network_delegate_helper_unittest.cc`

**Review by jumde — CHANGES_REQUESTED** [review]

**Review by iefremov — APPROVED** [review]
LGTM when the test is added

File: browser/net/brave_block_safebrowsing_urls.cc (hunk 1 above)
Comment (fmarier) [reply browser/net/brave_block_safebrowsing_urls.cc]: Added in the latest revision. Turns out I got the URL wrong. Good thing you asked for a test!

**Comment by kjozwiak** [comment]
Removing the `1.18.x - Release` milestone as this is still opened and hasn't landed in `master`. As per process, please don't associate unfished work to milestones.

**Comment by fmarier** [comment]
The cargo audit failure was fixed in https://github.com/brave/brave-core/pull/7853. Ignoring the CI failure for that.

**Review by jumde — APPROVED** [review]
lgtm

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.22.34 Chromium: 89.0.4389.48 (Official Build) nightly (64-bit)
-- | --
Revision | 0fe3c4589a6cf5ce719d167834dfa9cd8978937a-refs/branch-heads/4389@{#873}
OS | Windows 10 OS Version 2009 (Build 19042.804)
```

@brave/legacy_qa the valid STR/Test Plan is https://github.com/brave/brave-browser/issues/12297#issuecomment-775376042 and **NOT** the plan outlined via https://github.com/brave/brave-core/pull/6990#issue-512700429.

* installed `1.22.34 CR: 89.0.4389.48`
* waited till `https://safebrowsing.brave.com` downloaded the required files into `AppData\Local\BraveSoftware\Brave-Browser-Nightly\User Data\Safe Browsing`
* opened https://testsafebrowsing.appspot.com
* went through the `Webpage Warnings` section and ensured the malware warnings were being displayed
* went through the `Desktop Download Warnings` section and ensured that the Brave warned users via downloading potentially dangerous files
* ensured that selecting `Discard` via the downloads bar removed the download and removed the entry from `brave://downloads`
* ensured that selecting `Keep` via the downloads bar keeps the download without removing it
* ensured that selecting `Remove from list` via `brave://downloads` removes the entry/download
* ensured that selecting `Keep` via `brave://downloads` keeps the download without removing it
* ensured that clicking on `Learn More` opened https://support.brave.com/hc/en-us/articles/360018192491-How-do-I-fix-file-download-errors-?hl=en-US&ctx=3

discard1

discard2

**Pull request merged**